jenkinsfile-runner-steward-image/scripts/run.sh

```diff
@@ -162,6 +162,36 @@ function with_termination_log() {
   return "$rc"
 }
 
+function with_retries() {
+  local -r \
+    retry_interval=$1 \
+    timeout_seconds=$2 \
+    cmd=("${@:3}") \
+  #---
+
+  local rc
+
+  local start=$EPOCHSECONDS  # bash built-in
+
+  while true; do
+    rc=0; "${cmd[@]}" || rc=$?
+
+    (( rc != 0 )) || break
+
+    local elapsedseconds=$(( EPOCHSECONDS - start ))
+
+    if (( elapsedseconds > timeout_seconds )); then
+      printf "\nNot retrying anymore as timeout of %s seconds is reached.\n" "$timeout_seconds" >&2
+      break
```

Comment:
During testing we learnt that git-clone will hang for 130 seconds until connection times out if cloudci-git-mirror is down. In this case, with the timeout_seconds setting of 30 seconds there will be no retries. We will always end up in the "break" in line 185 after the first execution of a git-clone command. Is this correct?